### DISCUSSION TITLE
common vite config section

### DISCUSSION THREAD

**Discussion by apss-pohl** [alex8088/electron-vite#198] [Ideas]
### Clear and concise description of the problem

A common section for main/preload/render would be useful to prevent code duplication, e.g. when using shared alisases:

```ts
import { resolve } from 'node:path';
import { defineConfig, externalizeDepsPlugin } from 'electron-vite';


export default defineConfig({
  main: {
    resolve: {
      alias: {
        '@root': __dirname,
        '@m/': __dirname + '/apps/main/src/',
        '@r/': __dirname + '/apps/render/src/',
      },
    },
  },
  preload: {
    resolve: {
      alias: {
        '@root/': __dirname,
        '@m/': __dirname + '/apps/main/src/',
        '@r/': __dirname + '/apps/render/src/',
      },
    },
  },
  renderer: {
    resolve: {
      alias: {
        '@root/': __dirname,
        '@m/': __dirname + '/apps/main/src/',
        '@r/': __dirname + '/apps/render/src/',
      },
    },
  },
});
```

### Suggested solution

Following this structure:

```ts
export default {
  common: {
    // this applies to all sub configurations
  },
  main: {
    // vite config options
  },
  preload: {
    // vite config options
  },
  renderer: {
    // vite config options
  }
}
```

### Validations

- [X] Follow the [Code of Conduct](https://github.com/alex8088/electron-vite/blob/master/CODE_OF_CONDUCT.md).
- [X] Read the [Contributing Guidelines](https://github.com/alex8088/electron-vite/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://evite.netlify.app).
- [X] Check that there isn't [already an issue](https://github.com/alex8088/electron-vite/issues) that reports the same bug to avoid creating a duplicate.

**Comment by alex8088** [comment]
You can do it like this:

```js
import { resolve } from 'node:path';
import { defineConfig, externalizeDepsPlugin } from 'electron-vite';

const alias = {
  '@root': __dirname,
  '@m/': __dirname + '/apps/main/src/',
  '@r/': __dirname + '/apps/render/src/',
}

export default defineConfig({
  main: {
    resolve: {
      alias
    },
  },
  preload: {
    resolve: {
      alias
    },
  },
  renderer: {
    resolve: {
      alias
    },
  },
});

````

**Comment by apss-pohl** [comment]
Sure, this was more an example, I have a bunch of shared configuration. 

Therefore i am having 4 vite configuration files and i merge from the master file configurations used in renderer, main and preload. I am using lodash`s merge to bring them all together. Subscections configs overrule the main one.
Having a common section would look much cleaner instead of adding all of this in the subsections.

**Comment by alex8088** [comment]
I can understand your idea, but public configuration is easy to confuse and understand. Because the Vite configuration itself has many default configurations, there will be many conflicts when parsing each node configuration and merging with the public configuration.
